transaction/core/src/ring_signature/error.rs

```diff
@@ -2,13 +2,13 @@
 
 //! Errors which can occur in connection to ring signatures
 
+use crate::{range_proofs::error::Error as RangeProofError, TokenId};
+use alloc::string::{String, ToString};
 use displaydoc::Display;
 use serde::{Deserialize, Serialize};
 
 /// An error which can occur in connection to a ring signature
-#[derive(
-    Clone, Copy, Debug, Display, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize, Deserialize,
-)]
+#[derive(Clone, Debug, Deserialize, Display, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
 pub enum Error {
```

Comment:
Wow nice - love the error enumeration expansion to clearly capture the failure modes!

Reply:
yep, dont want to make it any harder to debug all this than it has to be :)